fe/fe-core/src/main/java/com/starrocks/mysql/privilege/DbPrivTable.java

```diff
@@ -67,17 +70,14 @@ public void getPrivs(UserIdentity currentUser, String db, PrivBitSet savedPrivs)
     /*
      * Check if user@host has specified privilege on any database
      */
-    public boolean hasPriv(String host, String user, PrivPredicate wanted) {
-        for (PrivEntry entry : entries) {
+    public boolean hasPriv(UserIdentity currentUser, PrivPredicate wanted) {
+        List<PrivEntry> userPrivEntryList = map.get(currentUser);
+        if (userPrivEntryList == null) {
+            return false;
+        }
+
+        for (PrivEntry entry : userPrivEntryList) {
             DbPrivEntry dbPrivEntry = (DbPrivEntry) entry;
-            // check host
-            if (!dbPrivEntry.isAnyHost() && !dbPrivEntry.getHostPattern().match(host)) {
-                continue;
-            }
-            // check user
-            if (!dbPrivEntry.isAnyUser() && !dbPrivEntry.getUserPattern().match(user)) {
-                continue;
-            }
             // check priv
```

Comment:
There may be a bug for the following case:
1. grant `zhangsan@%` to db1
2. grant `zhangsan@192.168.1.2` to db2
If I check the db1 priv for zhangsan, your pr will return false. Because the userIdentity is `zhangsan@192.168.1.2`.

Reply:
Thanks for your reminder. I've just uploaded a patch to fix that. Please take a look.